### PR TITLE
A complete rewrite of the README file

### PR DESCRIPTION
**Pull request by eladg**

### CODE REVIEW

**Comment by gaaady** [comment]
looks great

File: README.md [hunk 1]
```diff
@@ -1,108 +1,119 @@
...
 
-```ruby
-post '/big-reset' do
-  # assuming ENV['ADMIN_SECRET_TOKEN'] is correct, reset all queues
-end
+In order to get into the queue, a web client will submit a web form. Typically, the form will include the following data:
+```json
+{
+  "firstname": "John",
+  "firstname": "Doe",
```
Comment (eladg) [inline README.md]: `firstname` should be `lastname`

File: README.md [hunk 2]
```diff
@@ -1,108 +1,119 @@
...
+##### API Endpoints
+
+This is a simple [Sinatra](http://sinatrarb.com/) application. Please follow Sinatra's recommendation on how to deploy the app ([The Sinatra Book](http://sinatra-org-book.herokuapp.com/#toc_41)).
+
+For local development:
+- Make sure `bundle` is installed (if not follow [bundler.io](http://bundler.io/)).
+- Run `bundle install` to install dependencies.
+- Run server using: `bundle exec puma -p 3000 -C puma.rb`
+
+> **NOTICE**: use the `MIN_THREADS` & `MAX_THREADS` to set the number of puma threads to run on each api instance. For our need and since concurrency *was not* a concern with this simple solution & for our infrastructure, we found that each server may run very high (60+) amount of threads.
```
Comment (eladg) [inline README.md]: should be `MIN_THREADS` & `MAX_THREADS` env variables

File: README.md [hunk 3]
```diff
@@ -1,108 +1,119 @@
...
-`heroku run bundle exec rake midburn:close_queue --app midburn-queue`
+##### Set Env Variables
 
-2. Open the queue:
-`heroku run bundle exec rake midburn:open_queue  --app midburn-queue`
+For local development, we use the [dotenv gem](https://github.com/bkeepers/dotenv). copy `.env-example` to `.env` and configure the variables for your need.
 
-3. Getting the list of emails in the queue
-`heroku run bundle exec rake midburn:list --app midburn-queue`
+For production needs, follow your environment's best practices.
```
Comment (eladg) [inline README.md]: "... best practices for configuring env variables". For example, see Heroku's ["Configuration and Config Vars"](https://devcenter.heroku.com/articles/config-vars) & Amazon's EC2 ["Using Environment Variables"](http://docs.aws.amazon.com/opsworks/latest/userguide/apps-environment-vars.html).

File: README.md [hunk 4]
```diff
@@ -1,108 +1,119 @@
...
 ```
+For full list, see: [.env-example file!](./.env-example)
+
+> **NOTICE:** This list might not be up to date. Nevertheless, .env-example should always be updated to the most recent env changes.
+
+### Controlling the Queue
+
+To control a running instance of the queue, simply execute one of the following [rake](https://github.com/ruby/rake) tasks.
+
+> **NOTICE Heroku Users:** to execute a rake command on heroku, make sure you install [Heroku's Toolbelt](https://toolbelt.heroku.com/), login using: `heroku login` and then execute one of the above commands using: `heroku run`, For example: `heroku run bundle exec rake midburn:open_queue`
```
Comment (eladg) [inline README.md]: Should be triple ` instead of one.

**Comment by eladg** [comment]
thanks & of course it's not.

**Pull request merged**